src/types/redis_bitmap.cc

```diff
@@ -38,6 +39,22 @@ const char kErrBitmapStringOutOfRange[] =
     "The size of the bitmap string exceeds the "
     "configuration item max-bitmap-to-string-mb";
 
+// min_bytes can not more than kBitmapSegmentBytes
+void ExpandBitmapSegment(std::string *segment, size_t min_bytes) {
+  auto old_size = segment->size();
+  if (min_bytes > old_size) {
+    size_t expand_size = 0;
+    if (min_bytes > old_size * 2) {
+      expand_size = min_bytes - old_size;
+    } else if (old_size * 2 > kBitmapSegmentBytes) {
+      expand_size = kBitmapSegmentBytes - old_size;
```

Comment:
I think we need to check or dcheck `kBitmapSegmentBytes` and size here (as an separate helper function, more checks seems better)